charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -48,6 +52,11 @@ Selector labels
 {{- define "opentelemetry-collector.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "opentelemetry-collector.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
+{{- if eq .Values.mode "daemonset" }}
+app.kubernetes.io/component: agent-collector
```

Comment:
@dmitryax if this is changed from `app.kubernetes.io/component` to `component` it wouldn't break upgrades right?  But @nlamirault is this related to the ability to add additional labels?  Feels like we could keep hard coding `component` in the templates and add the ability to provide additional labels

Reply:
Also a side affect of adding `component` here would be that it is included in the templates' metadata labels.   Is that intended?

Reply:
Any change to selector labels breaks the k8s deployment upgrade. deployment has to be recreated.

Reply:
> @dmitryax if this is changed from app.kubernetes.io/component to component it wouldn't break upgrades right?

Should be right

Reply:
i change `component` to `app.kubernetes.io/component` to be consistent with [kubernetes recommanded labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/).
Actually, i could also add a `component` label into in the templates' metadata labels to not perform a breaking change

Reply:
@nlamirault is the change necessary to meet the problem you are trying to solve?  Isn't the addition of `additionalLabels` meeting your needs?

Reply:
@nlamirault see my question above

Reply:
We execute some policies on manifests, like check that Kubernetes recommanded labels exists.
It fails due to missing `app.kubernetes.io/component` label.
So i change the `component`  label to the Kubernetes recommanded label `app.kubernetes.io/component` 

Additional labels is another feature we would like.
To add to all resources a label like `mycompany/stack: observability`

Reply:
> Additional labels is another feature we would like.

Can you breakout this feature into its own PR?  

For `app.kubernetes.io/component` I think adding it in addition to `component` is an acceptable solution and works towards maybe dropping `component` in the future at some point.  @dmitryax do you agree?

Reply:
Any change in pod labels that are used in `matchLabels` section requires removing and creating the whole deployment. This is a significant breaking change for existing users, they would have to run `helm delete` and `helm install` to upgrade to the new version. I'm not saying I'm against it, but this is not how it should be handled. To make this through, this PR needs at least the following:
- Clear justification for the change with an issue submitted that includes references
- Documented upgrade plan for existing users. Ideally somehow automated, I'm sure we are not the only ones who want to handle this gracefully, so just look around for inspiration in other charts